h2o-core/src/main/java/water/UDPClientEvent.java

```diff
@@ -14,19 +14,24 @@
 
   @Override
   AutoBuffer call(AutoBuffer ab) {
-    // Handle only by non-client nodes
 
+    ClientEvent ce = new ClientEvent().read(ab);
     // Ignore messages from different cloud
-    if(ab._h2o._heartbeat._cloud_name_hash != H2O.SELF._heartbeat._cloud_name_hash) {
+    if (ce.senderHeartBeat._cloud_name_hash != H2O.SELF._heartbeat._cloud_name_hash) {
```

Comment:
This was crucial. We didn't actually report the client because `ab._h2o._heartbeat` was initiated to default java values. This we thought this node belongs to different cluster and we didn't continue.

This fix together with making sure that client is using at the start just a single H2O node make sure this is fixed. We don't even need confirm that the client has been distributed. At the start the client doesn't know about the nodes (except the one for initial communication). We distribute the client to the rest of the nodes and these nodes initially contact the client. 

The client waits for the cluster of the correct size and once the cloud is of the correct size, we can always be sure that the client is known on all nodes discovered by the client.